cms/server/AdminWebServer.py

```diff
@@ -523,6 +525,330 @@ def get(self, contest_id):
+                         .all():
+                t.num -= 1
+
+        elif operation == "Move up":
+            task2 = self.sql_session.query(Task)\
+                        .filter(Task.contest == self.contest)\
+                        .filter(Task.num == task.num - 1)\
+                        .first()
+
+        elif operation == "Move down":
+            task2 = self.sql_session.query(Task)\
+                        .filter(Task.contest == self.contest)\
+                        .filter(Task.num == task.num + 1)\
+                        .first()
+
+        # Swap task.num and task2.num, if needed
+        if task2 is not None:
+            task.num, task2.num = task2.num, task.num
+
+        if try_commit(self.sql_session, self):
+            # Create the user on RWS.
+            self.application.service.proxy_service.reinitialize()
+
+        # Maybe they'll want to do this again (for another task)
+        self.redirect(fallback_page)
+
+
+class ContestUserlistHandler(BaseHandler):
+    def get(self, contest_id):
+        self.contest = self.safe_get_item(Contest, contest_id)
+
+        self.r_params = self.render_params()
+        self.r_params["contest"] = self.contest
+        self.r_params["unassigned_users"] = \
+            self.sql_session.query(User)\
+                .filter(User.id.notin_(
+                    self.sql_session.query(Participation.user_id)
+                        .filter(Participation.contest == self.contest)
+                        .all()))\
+                .all()
+        self.render("contest_userlist.html", **self.r_params)
+
+
+class EditContestUserHandler(BaseHandler):
+    def post(self, contest_id):
+        fallback_page = "/contest/%s/users" % contest_id
+
+        self.contest = self.safe_get_item(Contest, contest_id)
+
+        try:
+            user_id = self.get_argument("user_id")
+            operation = self.get_argument("operation")
+            assert operation in (
+                "Remove from contest",
+            ), "Please select a valid operation"
+        except Exception as error:
+            self.application.service.add_notification(
+                make_datetime(), "Invalid field(s)", repr(error))
+            self.redirect(fallback_page)
+            return
+
+        user = self.safe_get_item(User, user_id)
+
+        if operation == "Remove from contest":
+            # Unassign the user from the contest.
+            participation = self.sql_session.query(Participation)\
+                .filter(Participation.user == user)\
+                .filter(Participation.contest == self.contest)\
+                .first()
+            self.sql_session.delete(participation)
+
+        if try_commit(self.sql_session, self):
+            # Create the user on RWS.
+            self.application.service.proxy_service.reinitialize()
+
+        # Maybe they'll want to do this again (for another task)
+        self.redirect(fallback_page)
+
+
+class TasklistHandler(BaseHandler):
```

Comment:
If I'm not mistaken, this and the next can be a SimpleContestHandler(something).